resources/server/marketplace/marketplace.db.ts

```diff
@@ -23,44 +22,42 @@ export class _MarketplaceDB {
       listing.description,
     ]);
 
-    const resultCast = result as ResultSetHeader;
-
-    return resultCast.insertId;
+    return insertId;
   }
 
   async fetchListings(): Promise<MarketplaceListing[]> {
     const query = 'SELECT * FROM npwd_marketplace_listings ORDER BY id DESC';
 
-    const [results] = await DbInterface._rawExec(query);
+    const results = await DbInterface.fetch(query);
     return <MarketplaceListing[]>results;
   }
 
   async deleteListing(listingId: number, identifier: string): Promise<void> {
     const query = 'DELETE FROM npwd_marketplace_listings WHERE id = ? AND identifier = ?';
 
-    await DbInterface._rawExec(query, [listingId, identifier]);
+    await DbInterface.exec(query, [listingId, identifier]);
   }
 
   async deleteListingsOnDrop(identifier: string) {
     const query = `DELETE FROM npwd_marketplace_listings WHERE identifier = ?`;
-    await DbInterface._rawExec(query, [identifier]);
+    await DbInterface.exec(query, [identifier]);
   }
 
   async getListing(listingId: number): Promise<MarketplaceListing> {
     const query = `SELECT * FROM npwd_marketplace_listings WHERE id = ?`;
-    const [results] = await DbInterface._rawExec(query, [listingId]);
+    const results = await DbInterface.fetch(query, [listingId]);
     const listings = <MarketplaceListing[]>results;
```

Comment:
As per last review comment, this method would benefit from a passable generic type here so we don't need to assert the result.